src/emoji-row.jsx

```diff
@@ -0,0 +1,43 @@
+import React, {Component, PropTypes} from 'react';
+import shallowCompare from 'react-addons-shallow-compare';
+import Emoji from './emoji';
+
+export default class EmojiRow extends Component {
+  static propTypes = {
+    emojis: PropTypes.array.isRequired,
+    onChange: PropTypes.func.isRequired,
+    style: PropTypes.object.isRequired,
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState);
+  }
+
+  render() {
+    const {emojis, onChange, style} = this.props;
+
+    return (
+      <div className="emoji-row" style={style}>
+        {emojis.map(emoji => {
+          return (
+            <Emoji
+              {...emoji}
+              ariaLabel={emoji.name}
+              role="option"
+              key={emoji.unicode}
+              onClick={function(e) {
+                onChange(emoji);
```

Comment:
One way would be to only pass the `onChange` callback to `Emoji`. And inside emoji do what you need to do, while binding the function only once in the constructor. (Same thing for the keyUp).